gerrit_python_tools/gerrit.py

```diff
@@ -624,13 +682,30 @@ def send_upstream(self, downstream, upstream):
         approvals = self.get_approvals(downstream.SSH())
 
         # Check to see if comment has necessary approvals.
-        if not self.is_upstream_approved(approvals):
-            msg = ("Could not send to upstream: One or more labels"
-                   " not approved.")
+        if self.is_forced():
+            if not self.is_release_approved(approvals):
+                msg = "Could not send to upstream: Release not approved."
+                logger.debug("Change %s: %s" % (self.change_id, msg))
+                ssh.exec_once('gerrit review -m %s %s'
+                              % (pipes.quote(msg), self.revision))
+                return
+            msg = ("***WARNING***\n%s (%s) has requested a forced upstream "
+                   "push. Bypassing all votes except for Release...\n\n"
+                   "Current votes are:\n\n%s"
+                   % (self._data['author']['name'],
+                      self._data['author']['email'],
+                      self.stringify_approvals(approvals)))
             logger.debug("Change %s: %s" % (self.change_id, msg))
             ssh.exec_once('gerrit review -m %s %s'
                           % (pipes.quote(msg), self.revision))
-            return
+        else:
+            if not self.is_upstream_approved(approvals):
```

Comment:
Could you please check for release approval here for this logging?? We should/would/could not expect Zuul to push upstream if release isn't approved. That's pretty clear if anyone's looking at the patch.